pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -1719,11 +1755,17 @@ private Set<String> findConsumingSegments(IdealState idealState) {
    *   - the value of `isTablePaused` flag in ideal state
    *   - list of consuming segments
    */
-  public PauseStatus getPauseStatus(String tableNameWithType) {
+  public PauseStatusDetails getPauseStatus(String tableNameWithType) {
     IdealState idealState = getIdealState(tableNameWithType);
-    String isTablePausedStr = idealState.getRecord().getSimpleField(IS_TABLE_PAUSED);
     Set<String> consumingSegments = findConsumingSegments(idealState);
-    return new PauseStatus(Boolean.parseBoolean(isTablePausedStr), consumingSegments, null);
+    PauseState pauseStatus = extractTablePauseStatus(idealState);
+    if (pauseStatus != null) {
+      return new PauseStatusDetails(pauseStatus.isPaused(), consumingSegments, pauseStatus.getReasonCode(),
+          pauseStatus.getComment(), pauseStatus.getTimeInMillis());
+    }
+    String isTablePausedStr = idealState.getRecord().getSimpleField(IS_TABLE_PAUSED);
+    return new PauseStatusDetails(Boolean.parseBoolean(isTablePausedStr), consumingSegments,
+        PauseState.ReasonCode.ADMINISTRATIVE, null, new Timestamp(System.currentTimeMillis()).toString());
```

Comment:
Let's use an empty timestamp for the old pauses (using isTablePaused flag) so it's not misleading.